api/v1alpha1/addresspool_conversion.go

```diff
@@ -0,0 +1,45 @@
+package v1alpha1
+
+import (
+	"github.com/metallb/metallb-operator/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this AddressPool to the Hub version (vbeta1).
+func (src *AddressPool) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta1.AddressPool)
+	dst.Spec.Protocol = src.Spec.Protocol
+	dst.Spec.Addresses = make([]string, len(src.Spec.Addresses))
+	copy(dst.Spec.Addresses, src.Spec.Addresses)
+	dst.Spec.AutoAssign = src.Spec.AutoAssign
```

Comment:
do u meant to assign the pointers or the content ?
`*dst.Spec.AutoAssign = *src.Spec.AutoAssign`

Reply:
Good catch, need to change that I think.